util/headerreader/header_reader.go

```diff
@@ -91,25 +94,25 @@ var TestConfig = Config{
 	UseFinalityData:  false,
 }
 
-func New(ctx context.Context, client arbutil.L1Interface, config ConfigFetcher) (*HeaderReader, error) {
+func New(ctx context.Context, client arbutil.L1Interface, config ConfigFetcher, arbSysPrecompile ArbSysInterface, usePrecompilesgen bool) (*HeaderReader, error) {
```

Comment:
Instead of `usePrecompilesgen` could we just check `if arbSysPrecompile != nil`?

Reply:
Thats what I did before, but this breaks in loadbalancer since it tries to fetch the codeAt types.ArbSysAddress and this returns an error. 

one way to fix that would be to move the entire code from checking codeAt to generating arbSys outside of headerreader.New and have the above check `if arbSysPrecompile != nil` to set `isParentChainArbitrum`